usaspending_api/database_scripts/job_archive/management/commands/dev_4109_repair_awards.py

```diff
@@ -0,0 +1,169 @@
+"""
+Jira Ticket Number(s): DEV-4109
+
+Expected CLI:
+
+    $ python3 manage.py dev_4109_repair_awards
+
+Purpose:
+
+    Correct Faulty FKs and missing award records
+"""
+
+import logging
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+from usaspending_api.common.helpers.timing_helpers import Timer
+
+logger = logging.getLogger("console")
+
+CHECK_MISSING_AWARDS = """
+    SELECT tf.{field}
+    FROM {table} AS tf
+    LEFT OUTER JOIN awards a on a.generated_unique_award_id = tf.unique_award_key
+    WHERE a.id IS NULL;
+"""
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--inspect",
+            action="store_true",
+            help=(
+                "Don't execute DML commands. "
+                "Print results of SQL statements to review database "
+                " records to track data state over time"
+            ),
+        )
+
+    def handle(self, *args, **options):
+
+        if options["inspect"]:
+            self.inspect()
+            return
+
+        with Timer(f"{__name__}"):
+            self.part_one()
+            self.part_two()
+            self.part_three()
+
+    def part_one(self):
+        with Timer("Part 1: Addressing Missing Award Records"):
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    CHECK_MISSING_AWARDS.format(table="transaction_fpds", field="detached_award_procurement_id")
+                )
+                ids_to_load = [str(row[0]) for row in cursor.fetchall()]
+
+            if ids_to_load:
+                logger.info(f"Reloading {len(ids_to_load)} transactions from Broker")
+                call_command("load_fpds_transactions", "--ids", " ".join(ids_to_load))
+            else:
+                logger.info("No transactions to load")
+
+    def part_two(self):
+        fix_txn_fk_to_awards = """
+            UPDATE transaction_normalized AS tn
+            SET award_id = a.id
+            FROM awards a
+            WHERE
+                tn.unique_award_key = a.generated_unique_award_id
+                AND tn.award_id IS DISTINCT FROM a.id
+            RETURNING tn.id
+        """
+        with Timer("Running Part 2: Fixing Transaction FKs"):
+            with connection.cursor() as cursor:
+                cursor.execute(fix_txn_fk_to_awards)
+                transactions = [str(row[0]) for row in cursor.fetchall()]
+            if transactions:
+                logger.info(f"Updated transactions:\n{', '.join(transactions)}")
+
+    def part_three(self):
+        logger.info("Part 3: Re-Computing Awards")
+        logger.warning("Run usaspending-api/usaspending_api/scripts/archive/recompute_all_awards.py")
```

Comment:
It doesn't actually run the command?

Reply:
I hope that is a rhetorical question.

Reply:
I was asking if it was run somewhere else. It seems strange that part three of the script is to do absolutely nothing.

Reply:
Log statements are "absolutely nothing" now?

The script it is directing someone to run is pre-existing and the run last time took over 24hrs. While it could be called by this script the benefit seems minimal at best. The context for these scripts are to correct data and are manually run on an ad-hoc basis. These log statements are code documentation in case the operations ticket is lost or not easily found.

Reply:
I misinterpreted that message as a statement of what the script was currently doing. Technically the future tense hints at that, but you might want to say "please run" or something to be more clear. Since it's a short-lived script I'm less worried about this since future generations of devs will (hopefully) never see this.